### ISSUE TITLE
Warning: Illegal offset type in isset or empty in UnitOfWork

### ISSUE THREAD

**Issue by astalker** [doctrine/doctrine2#6193]
I'm experiencing the following error `Warning: Illegal offset type in isset or empty` from UnitOfWork.php. I believe this was introduced in v2.5.5 and is still apparent in v2.5.6

Thrown from the following line 3469 in v2.5.5 (3486 in v2.5.6).

```
        if (! isset($this->identityMap[$entityName])) {
            return;
        }
```

This is caused when passing an entityName to the clear method such as 
`$em->clear($entity);`

Please let me know if further details are required.

**Comment by Ocramius** [comment]
`$entity` must be a string.

This is a duplicate of https://github.com/doctrine/doctrine2/pull/6017

**Comment by astalker** [comment]
Cool, I had a look through but wasn't sure if it was exactly the same issue or not.

**Comment by astalker** [comment]
@Ocramius I've looked at the changes in #6017 and it still causes an issue for us as it enforces the existing behaviour with another error. Essentially the behaviour change in 2.5.5 is still present.

I understand we are passing in the wrong thing essentially but its not an insignificant task for us to go through and make all the changes as a result of a bug fix release. For now I've updated our dependencies to use 2.5.4. We will of course make the correct changes going forward.

Its not a big deal in this case as we caught it in time but I think it important to highlight with a bit of feedback. A minor behaviour change to enforce correct behaviour can have much larger knock on effects when included as a bug fix release rather than a minor or major release with BC implications highlighted.

The work is appreciated of course - cheers and Merry Christmas!
